lib/RemoveVariableEventBehaviour.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+var inherits = require('inherits');
+
+var CommandInterceptor = require('diagram-js/lib/command/CommandInterceptor').default;
+var is = require('bpmn-js/lib/util/ModelUtil').is;
+var getBusinessObject = require('bpmn-js/lib/util/ModelUtil').getBusinessObject;
+
+/**
+ * Remove 'camunda:variableEvents' property when a startEvent is moved out of an event subProcess
+ */
+function RemoveVariableEventBehaviour(
+    modeling, injector, bpmnFactory, moddleCopy
+) {
+
+  injector.invoke(CommandInterceptor, this);
+
+  this.postExecuted(['shape.move'], function(context) {
+
+    var oldParent = context.oldParent;
+    var oldParentBusinessObject = getBusinessObject(oldParent);
+    var newParent = context.newParent;
+    var shape = context.shape;
+    var shapeBusinessObject = getBusinessObject(shape);
+    var eventDefinitions, definitionsCopy;
+
+    // if shape is a startEvent
+    if (is(shape, 'bpmn:StartEvent')) {
+
+      eventDefinitions = shapeBusinessObject.get('eventDefinitions');
+      definitionsCopy = eventDefinitions.slice();
+
+      definitionsCopy.forEach(function(def,i) {
+
+        var newDef = bpmnFactory.create('bpmn:ConditionalEventDefinition');
+        moddleCopy.copyElement(def, newDef);
+        newDef.$parent = def.$parent;
+
+        // if event definitions have variableEvents property
+        if (def.get('camunda:variableEvents')) {
+
+          // if shape within an event subpprocess changes parent
+          if (
+            is(oldParent, 'bpmn:SubProcess') &&
+            oldParentBusinessObject.get('triggeredByEvent') &&
+            !is(newParent, 'bpmn:SubProcess')
+          ) {
+
+            // remove variableEvents property
+            newDef.variableEvents = undefined;
+            definitionsCopy[i] = newDef;
+
+          }
+        }
+      });
+    }
+
+    modeling.updateProperties(shape, { 'eventDefinitions': definitionsCopy });
```

Comment:
We want to update event definitions only when it's necessary. As it is here in the code, the `eventDefinitions` property will be updated whenever one moves any shape as there are no conditions applied.

Reply:
Also, let's not update the properties if no `camunda:variableEvents` had to be replaced.